src/components/login/SignIn.tsx

```diff
@@ -7,27 +7,37 @@ import {
   InputGroup,
   InputLeftElement,
   Text,
-} from '@chakra-ui/react';
-import { AxiosError } from 'axios';
-import { useState } from 'react';
-import { AiFillLock, AiOutlineMail } from 'react-icons/ai';
-import { Flex } from '../primitives/Flex';
-import { useAuth } from '../../contexts/auth';
-import { theme } from '../../theme';
-import { ChangeLoginMode } from './ChangeLoginMode';
+  useToast,
+} from "@chakra-ui/react";
+import { AxiosError } from "axios";
+import { useState, useRef } from "react";
+import { AiFillLock, AiOutlineMail } from "react-icons/ai";
+import { Flex } from "../primitives/Flex";
+import { useAuth } from "../../contexts/auth";
+import { theme } from "../../theme";
+import { ChangeLoginMode } from "./ChangeLoginMode";
 
 type SignInProps = { changeMode: () => void };
 export const SignIn = ({ changeMode }: SignInProps) => {
-  const [email, setEmail] = useState('');
-  const [password, setPassword] = useState('');
-  const [error, setError] = useState('');
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [error, setError] = useState("");
   const [isLoading, setLoading] = useState(false);
   const { authenticate } = useAuth();
+  const toast = useToast();
+  const toastIdRef = useRef();
+
+  function closeToast() {
+    if (toastIdRef.current) {
+      toast.close(toastIdRef.current);
+    }
+  };
 
```

Comment:
rather for arrow functions.
```suggestion
  const closeToast = () => {
    if (toastIdRef.current) {
      toast.close(toastIdRef.current);
    }
  };